stl/inc/coroutine

```diff
@@ -0,0 +1,337 @@
+// coroutine standard header (core)
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _COROUTINE_
+#define _COROUTINE_
+#include <yvals_core.h>
+#if _STL_COMPILER_PREPROCESSOR
+#if _HAS_CXX20
+#ifdef __cpp_coroutines
+#include <type_traits>
+
+#pragma pack(push, _CRT_PACKING)
+#pragma warning(push, _STL_WARNING_LEVEL)
+#pragma warning(disable : _STL_DISABLED_WARNINGS)
+_STL_DISABLE_CLANG_WARNINGS
+#pragma push_macro("new")
+#undef new
+
+// LWG TODO:
+// * The implicit correspondence between `address() != nullptr` and "the coroutine handle refers to a coroutine" needs
+//   to be made explicit.
+// * Should `coroutine_handle<>`'s comparison operators be hidden friends?
+// * `noop_coroutine_handle` makes no sense after a slicing assignment. Maybe it should have a conversion to
+//   `coroutine_handle<>` rather than being publicly derived? (This interacts with the "hidden friend" comparison
+//   operator question.)
+// * [coroutine.trivial.awaitables]/1 needs a cross-reference for *await-expression*.
+//
+// * [coroutine.handle]
+//   * Replace para 1:
+//     An object whose type is a specialization of `coroutine_handle` is called a *coroutine handle* and can be used to
+//     refer to a suspended or executing coroutine. A coroutine handle refers to a coroutine if and only if the return
+//     value of `address()` is non-null.
+//   * Replace para 2:
+//     A program shall not declare an explicit or partial specialization of `coroutine_handle`.
+// * coroutine_handle<Promise>::from_promise:
+//   Expects: `p` references the promise object of a coroutine.
+//   Returns: A coroutine handle `h` that refers to the coroutine whose promise object is referenced by `p`. [Note:
+//            `addressof(h.promise()) == addressof(p)` is consequently `true`.-end note]
+//   Throws: Nothing.
+// * coroutine_handle<>::address:
+//   Returns: A non-null pointer value if and only if `*this` refers to a coroutine.
+// * coroutine_handle<>::from_address and coroutine_handle<Promise>::from_address:
+//   Expects: ...

```

Comment:
More LWG issue fodder: "*Requires*: `addr` was obtained via a prior call to `address`." is insufficient for `coroutine_handle<Promise>::from_address`; it must require that `addr` is a null pointer value or was obtained via `h.address()` for a coroutine handle `h` that refers to a coroutine whose promise object has type `Promise`.